pandas/core/groupby.py

```diff
@@ -4437,20 +4437,15 @@ def _reorder_by_uniques(uniques, labels):
 
 
 def _groupby_indices(values):
+    """
+    return a dictionary of groups -> indices
+    """
+    if not is_categorical_dtype(values):
+        values = Categorical(values)
```

Comment:
actually I see you proposed something else. ``.group_index`` is not defined except for a single ``Grouping``. If we had a ``MultiGrouping``, then yes I think that would work (right now that basically a list of ``Groupings``).